[rocthrust] Edit the reference section of the documentation. (#6145)
ROCm/rocm-libraries: projects/rocthrust/docs/conceptual/rocThrust-synchronization-and-blocking.rst

.. meta::
  :description: Definitions of synchronous and asynchronous behavior in rocThrust
  :keywords: rocThrust, ROCm, synchronous, asynchronous, blocking, host, device, HIP

.. _synchronization-and-blocking:

**********************************************

> Synchronization and blocking in rocThrust

Reviewer: ```suggestion
rocThrust synchronization and blocking
```

**********************************************
 

> Operations that are synchronous with respect to the host (CPU) are operations that run on the GPU and block the CPU. The CPU will remain idle until the operation completes on the GPU. Once the operation on the GPU finishes, the CPU will launch the next instruction. 

Reviewer: ```suggestion
Synchronous operations with respect to the host (CPU) run on the GPU and block the CPU. The CPU remains idle until the operation completes on the GPU. Once the GPU operation finishes, the CPU will launch the next instruction. 
```

Author: These changes make the passage inaccurate.

> Operations that are asynchronous with respect to the host are operations that don't block the CPU. The CPU can continue launching more operations and doing other work while the GPU runs its workload. The GPU and the CPU can run concurrently. The host and device must be synchronized before the host can access the results of the operations. 

Reviewer: ```suggestion
Asynchronous operations with respect to the host don't block the CPU. The CPU can continue launching more operations and doing other work while the GPU runs its workload. The GPU and the CPU can run concurrently. The host and device must be synchronized before the host can access the results of the operations. 
```

Author: This change would make it technically inaccurate as the term is actually "asynchronous wrt the host"

rocThrust functions are synchronous with respect to the host by default. 

> The ``hip_rocprim::par_nosync`` policy can be used to avoid synchronization barriers. This policy communicates to the runtime that synchronization isn't necessary and that it can be skipped if the implementation allows it. This policy doesn't guarantee non-blocking calls because the operations themselves could be blocking. There is no way to guarantee asynchronization in rocThrust.

Reviewer 2: I have a suggestion: maybe instead of "because some operations themselves could be blocking.", it might be clearer to say something like "because some algorithms require internal synchronization."
